cli/js/net.ts

```diff
@@ -109,63 +110,56 @@ export class ConnImpl implements Conn {
 }
 
 export class ListenerImpl implements Listener {
-  constructor(
-    readonly rid: number,
-    readonly addr: Addr,
-    private closing: boolean = false
-  ) {}
+  private closed = false;
+
+  constructor(readonly rid: number, readonly addr: Addr) {}
 
   async accept(): Promise<Conn> {
+    if (this.closed) {
+      // This should match the error thrown by op_accept if closing.
+      throw new errors.Other("Listener has been closed");
```

Comment:
This is brittle; I don't like that `Listener` and `Socket` change returned error to such string, while all other resources don't...

Reply:
We already had this kind of check in `next()`. It seems an `accept()`-after-`close()` throws `BadResource`, but a `close()`-while-`accept()` gives `Other: Listener has been closed`. This code coerces the first case into the second.

How about this: check for `BadResource` for the break condition in the iterator, and change the Rust error kind for `close()`-while-`accept()` to `BadResource`?

Reply:
We shouldn't even expose errors.Other - it maps to Error. I will create a PR to remove it.

Reply:
Removing Deno.errors.Other here https://github.com/denoland/deno/pull/4249